File: frontend/my_app/lib/models/journal_entry.dart
class JournalEntry {
  final int? id;
  final String content;
  final int? moodRating;
  final String? moodTags;
  final int? stressRating;
  final String? emotions;
  final DateTime? timestamp;

  JournalEntry({
    this.id,
    required this.content,
    this.moodRating,
    this.moodTags,
    this.stressRating,
    this.emotions,
    this.timestamp,
  });

  factory JournalEntry.fromJson(Map<String, dynamic> json) {
    return JournalEntry(
      id: json['id'],
      content: json['content'],
      moodRating: json['moodRating'],
      moodTags: json['moodTags'],
      stressRating: json['stressRating'],
      emotions: json['emotions'],
      timestamp: json['timestamp'] != null 
          ? DateTime.parse(json['timestamp']) 
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'content': content,
      'moodRating': moodRating,
      'moodTags': moodTags,
      'stressRating': stressRating,
      'emotions': emotions,
    };
  }
}

File: backend/src/main/java/com/aks/bc/backend/model/JournalEntry.java
package com.aks.bc.backend.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "journal_entries")
public class JournalEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(columnDefinition = "TEXT")
    private String content;

    private Integer moodRating;
    private String moodTags;

    private Integer stressRating;
    private String emotions;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime timestamp;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false,updatable = false)
    private User user;
}
